core/src/packets/icmp/v4/mod.rs

```diff
@@ -294,10 +303,16 @@ impl Icmpv4Payload for () {
 /// ## Remarks
 ///
 /// When using the associated derive macro, the payload struct implementation
-/// must provide an private implementation of the `cascade` function.
+/// must provide an private implementation of the `reconcile` function.
 ///
 /// [`Packet`]: crate::packets::Packet
-pub trait Icmpv4Packet<P: Icmpv4Payload>: Packet<Header = Icmpv4Header, Envelope = Ipv4> {
+pub trait Icmpv4Packet<P: Icmpv4Payload>: Packet<Envelope = Ipv4> {
+    /// Returns a reference to the header.
+    fn header(&self) -> &Icmpv4Header;
```

Comment:
just for comment, as we discussed this, but these "public" trait fns for header will eventually become privately encapsulated, right @drunkirishcoder? (in another PR)

Reply:
yeah trying to make icmp headers internal to the crate as well.